operator/pkg/lib/sidecars/pods/get.go

```diff
@@ -0,0 +1,151 @@
+package pods
+
+import (
+	"context"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+type SidecarImage struct {
+	Repository string
+	Tag        string
+}
+
+const (
+	istioValidationContainerName = "istio-validation"
+	istioInitContainerName       = "istio-init"
+	istioSidecarName             = "istio-proxy"
+)
+
+func (r SidecarImage) matchesImageIn(container v1.Container) bool {
+	// TODO Understand why we can do a full string match
+	containsRepository := strings.Contains(container.Image, r.Repository)
```

Comment:
This was used as we were using `solo.io` images at some point, but it may be possible to drop it now

Reply:
Agree, we can simplify it to check the tag only.
Another idea is to check what `istioctl version` is doing to get this version.